annotation-processor/src/main/java/org/hibernate/validator/ap/checks/MethodParameterOverrideCheck.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks;
+
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Checks, that the given parameter has a correct set of constraints if it's overriding a method form other class.
+ * Applies to methods parameters.
+ */
+public class MethodParameterOverrideCheck extends AbstractConstraintCheck {
+
+	private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+
+	private Elements elementUtils;
+
+	private Types typeUtils;
+
+	public MethodParameterOverrideCheck(Elements elementUtils, Types typeUtils) {
+		this.elementUtils = elementUtils;
+		this.typeUtils = typeUtils;
+	}
+
+	@Override
+	public Set<ConstraintCheckError> checkField(VariableElement element, AnnotationMirror annotation) {
+		// get a current class in which a method is located.
```

Comment:
Yes, that's a good question. I've started to implement this as checkMethod. But actually the annotations are placed not on the method but on the method parameters. So that's why I've added `ElementKind.PARAMETER` to `SingleValuedChecks` class. Because otherwise those methods with annotations on parameters were ignored.

Also those constraint sanity checks should be added to `parameterChecks` map in `ConstraintCheckFactory` when they are merged in to master. Or should I just merge that branch (HV-270) into this one so those checks are present in here as well ? 

Return value constraints should be already supported in HV-270, but parameter constraints - no. I didn't want to add the same code to support parameters in two different  branches, so I was thinking that when one of the PR will be merged, I'll update the other.